bootstrap/src/test.rs

```diff
@@ -0,0 +1,157 @@
+use std::fs::File;
+use std::path::{Path, PathBuf};
+
+use anstream::{eprint as print, eprintln as println};
+use clap::Args;
+use color_print::{cprint, cprintln};
+use glob::glob;
+use which::which;
+
+use crate::manifest::Manifest;
+
+/// Run tests
+#[derive(Args, Debug)]
+pub struct TestCommand {}
+
+impl TestCommand {
+    pub fn run(&self, manifest: &Manifest) {
+        manifest.prepare();
+
+        std::panic::set_hook(Box::new(|info| {
+            cprintln!("<r,s>Test failed</r,s>: {}", info);
+        }));
+
+        let testcases = self.collect_testcases(manifest);
+        cprintln!("<b>[TEST]</b> found {} testcases", testcases.len());
+
+        let filechecker = FileChecker::new();
+        for testcase in testcases {
+            match testcase.test {
+                TestType::FileCheck => {
+                    cprint!("File checking {}...", testcase.name);
+                    testcase.build(manifest);
+                    filechecker.run(&testcase.source, &testcase.output);
+                }
+                TestType::Compile => {
+                    cprint!("Compiling {}...", testcase.name);
+                    testcase.build(manifest);
+                }
+            }
+            cprintln!("<g>OK</g>");
+        }
+    }
+
+    pub fn collect_testcases(&self, manifest: &Manifest) -> Vec<TestCase> {
+        let mut result = vec![];
+
+        // Examples
+        for case in glob("example/*.rs").unwrap() {
+            let case = case.unwrap();
+            let filename = case.file_stem().unwrap();
+            if filename == "mini_core" {
+                continue;
+            }
+            let name = format!("example/{}", filename.to_string_lossy());
```

Comment:
Remark: you might want to use `Path::join` functionality so it gets proper platform-specific separators if you want this codegen to support not just Unixes.

Reply:
This is merely a display name, not the actual path. In fact, it will consist of only two components: *example/xxx*. Therefore, I believe it is unnecessary to manage separators in a platform-specific manner.

Reply:
That's fair enough.